eng/scripts/Update-Mgmt-CI.ps1

```diff
@@ -9,38 +9,41 @@ function Update-CIFile() {
 
     $shouldRemove = $false
 
-    $lines = Get-Content $mgmtCiFile
-    $newLines = [System.Collections.ArrayList]::new()
-    foreach($line in $lines) {
-        if($line.StartsWith("    ArtifactName:")) {
-            Continue
-        }
-        if($line.StartsWith("trigger:") -or $line.StartsWith("pr:")) {
-            $prefix = $line.Substring(0, $line.IndexOf(":") + 1)
-            $newLines.Add("$prefix none") | Out-Null
-            $shouldRemove = $true
-            Continue
-        }
-        if($line.StartsWith("    - name:")) {
-            if($line -match "Azure.ResourceManager.") {
-                $shouldRemove = $false
-            }
-            else {
-                $shouldRemove = $true
-                Continue
-            }
-        }
-        if($shouldRemove) {
-            if($line.StartsWith(" ")) {
-                Continue
-            }
-            $shouldRemove = $false
-        }
+    $content = Get-Content $mgmtCiFile -Raw
 
-        $newLines.Add($line) | Out-Null
+    if ($content -match "(?s)ServiceDirectory: (?<sd>[^\r\n]+).*- name: (?<p>[^\r\n]+)")
```

Comment:
I assume the templating makes this consistent, but is it worth handling whitespace more flexibly (e.g. if someone adds two spaces)?

Reply:
These regex's are definitely hacky but right now there isn't any good yml parser we can use to maintain formatting. For these consistency should be kept by the fact that these are mostly generated. That said I probably should add some \s+ instead of hardcoding exacts. The stupid multiline matching is kind of a pain though because \s also matches newlines.